pretext/project/__init__.py

```diff
@@ -1668,6 +1670,133 @@ def deploy(
         if not stage_only:
             utils.publish_to_ghpages(self.stage_abspath(), update_source)
 
+    def is_git_managed(self) -> bool:
+        return (self.abspath() / ".git").exists()
+
+    def update_boilerplate(self, backup: bool = False, force: bool = False) -> None:
+        """
+        Checks each of the managed files in a project.  If the file matches the
+        hash of the default file for the provided version, the file will be updated
+        to the newest version.  If it does not match, this means the user modified it.
+        In that case, we warn, add a note to the top of the file, and continue.
+
+        The managed files are stored in constants.PROJECT_RESOURCES:
+        - project.ptx
+        - requirements.txt
+        - .gitignore
+        - .devcontainer.json
+        - .github/workflows/pretext-cli.yml
+        - codechat_config.yaml
+        """
+        # Get hash of default files:
+        with importlib.resources.open_text(
```

Comment:
Consider whether `pretext.resources` should be used instead somehow

Reply:
Done